Utils/Editor/Tests/AsyncTaskListTest.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+
+namespace DUCK.Utils.Editor.Tests
+{
+	[TestFixture]
+	public class AsyncTaskListTest
+	{
+		[Test]
+		public void ExpectConstructorToNotThrow()
+		{
+			Assert.DoesNotThrow(() =>
+			{
+				var tasks = new AsyncTaskList();
+			});
+		}
+
+		[Test]
+		public void ExpectAddNotToThrow()
+		{
+			Assert.DoesNotThrow(() =>
+			{
+				var tasks = new AsyncTaskList();
+				tasks.Add(() => { });
+			});
+		}
+
+		[Test]
+		public void ExpectSyncTaskToBeCalled()
+		{
+			var wasCalled = false;
+
+			var tasks = new AsyncTaskList();
+			tasks.Add(() => { wasCalled = true; });
+			tasks.Run();
+
+			Assert.IsTrue(wasCalled);
+		}
+
+		[Test]
+		public void ExpectAsyncTaskToBeCalled()
+		{
+			var wasCalled = false;
+
+			var tasks = new AsyncTaskList();
+			tasks.Add((next) =>
+			{
+				wasCalled = true;
+				next();
+			});
+			tasks.Run();
```

Comment:
Probably worth doing an `Assert.IsTrue(wasCalled);` just before the task run to prove that this isn't a false positive.

I.e. if someone changed Add to immediately execute this test may still pass but it Add wouldn't be expected to act that way.

Reply:
Will do